### ISSUE TITLE
Large bundle size on main thread when using the worker

### ISSUE THREAD

**Issue by mousetail** [val-town/codemirror-ts#35]
# Issue

When building a project with Vite, both the "worker" and "main" threads generate very large bundles, presumably including all of typescript:
```
static/target/assets/typescript_worker-CanckLWI.js  3,545.27 kB
static/target/assets/index-DGW34Qfz.js                489.24 kB │ gzip:   164.90 kB
static/target/assets/index-DhtVGfxZ.js              3,530.38 kB │ gzip: 1,010.25 kB
```

# Reproduction Steps

I have the following minimal setup:
```ts
async function initTypescriptForCodebox(): Promise<typeof minimalSetup> {
    const { tsFacetWorker, tsSyncWorker, tsLinterWorker, tsAutocompleteWorker, tsHoverWorker } = await import("@valtown/codemirror-ts");

    if (typeScriptEnvironment === undefined) {
        const innerWorker = new Worker(new URL("./typescript_worker.ts", import.meta.url), {
            type: "module",
        });
        const worker = Comlink.wrap(innerWorker) as WorkerShape;
        await worker.initialize();
        typeScriptEnvironment = worker;
    }
    const path = '/src/index.ts'

    return [
        basicSetup,
        javascript({
            typescript: true,
            jsx: true,
        }),
        tsFacetWorker.of({ worker: typeScriptEnvironment, path }),
        tsSyncWorker(),
        tsLinterWorker(),
        autocompletion({
            override: [tsAutocompleteWorker()],
        }),
        tsHoverWorker()
    ]
}
```

The only import from typescript is via codemirror-ts. Since I'm using worker mode I expect this to load only some basic types, but it seems it's loading the entirety of Typescript into both bundles.

**Comment by tmcw** [comment]
Should be fixed with #36 - added `"sideEffects": false` to our `package.json`.

**Comment by tmcw** [comment]
Fixed with #36

**Comment by olsonpm** [comment]
I'm running into this as well, mind explaining how you're tree-shaking this ?  I'm still running into a large bundle size when requiring codemirror-ts in a next.js project.

And separately testing this with rollup, the bundle is still 9Mb

<details>
<summary>code</summary>

with config
```js
// rollup.config.js
import { nodeResolve } from '@rollup/plugin-node-resolve'
import commonjs from '@rollup/plugin-commonjs'

export default {
  input: './entry.mjs',
  output: {
    dir: './bundle',
  },
  plugins: [nodeResolve(), commonjs()],
}
```

and entry
```js
// entry.mjs
const { tsAutocompleteWorker } = await import('@valtown/codemirror-ts')

console.log(tsAutocompleteWorker)
```

results in
```sh
❯ ls -lh bundle
total 9.0M
... 106 ... entry.js
... 9.0M ... index-Rm347p5j.js
```

</details>
